contracts/vault/AssetManagement.sol

```diff
@@ -191,6 +191,11 @@ abstract contract AssetManagement is
 
         uint256 amountToWithdraw = amount;
 
+        if (amount < (type(uint256).max - 1e18)) {
```

Comment:
do we need this? 
After all, in practice, we will never have so much money that the amount would be that large.
You can just add a comment that this value will never be too large.

Reply:
A better solution would be to change this maximum value in the function to something smaller and always add it here instead of using an additional if statement.
```
    function withdrawAll()
        external
        override
        whenNotPaused
        onlyAmmTreasury
        returns (uint256 withdrawnAmount, uint256 vaultBalance)
    {
        return _withdraw(type(uint256).max);
    }
```

Reply:
Done.